internal/core/integrations/gitlab_integration_test.go

```diff
@@ -0,0 +1,105 @@
+package integrations
+
+import (
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+
+	"github.com/stretchr/testify/assert"
+	gitlab "gitlab.com/gitlab-org/api/client-go"
+)
+
+func TestCreateProjectHook(t *testing.T) {
+	t.Run("Returned ProjectHookOption Struct should have the URL set to main devguard", func(t *testing.T) {
+		os.Setenv("INSTANCE_DOMAIN", "https://api.main.devguard.org")
+
+		hooks := []*gitlab.ProjectHook{}
+		token, err := uuid.NewUUID()
+		if err != nil {
+			slog.Error("error when trying to generate token")
+			return
+		}
+		results, err := CreateProjectHookOptions(token, hooks)
+		if err != nil {
+			slog.Error(err.Error())
+			return
+		}
+		assert.Equal(t, "https://api.main.devguard.org", *results.URL)
+
+	})
+	t.Run("Returned ProjectHookOption Struct should have the URL set to stage devguard", func(t *testing.T) {
+
+		os.Setenv("INSTANCE_DOMAIN", "https://api.stage.devguard.org")
+
+		hooks := []*gitlab.ProjectHook{}
+		token, err := uuid.NewUUID()
+		if err != nil {
+			slog.Error("error when trying to generate token")
+			return
+		}
+		results, err := CreateProjectHookOptions(token, hooks)
+		if err != nil {
+			slog.Error(err.Error())
+			return
+		}
+		assert.Equal(t, "https://api.stage.devguard.org", *results.URL)
```

Comment:
We need the whole webhook url (including the path) to be added. This applies to all tests
```suggestion
		assert.Equal(t, "https://api.stage.devguard.org/api/v1/webhook/", *results.URL)
```